### PR TITLE
[WIP] add HIDL server support

### PR DESCRIPTION
**Pull request by zhanglongxia**
This CL implements the HIDL server on the Android platform. The HIDL
server provides IThread and IThreadSettings HIDL interfaces for the
HIDL client.
The IThread interface is implemented by the HIDL server to provides
all control operations of the Thread network.
The IThreadSettings interface is mainly called by the HIDL server to
send sensitive settings to Thread System service. The Thread System
servie stores the sensitive settings to a secure area.

### CODE REVIEW

File: src/common/types.cpp [hunk 1]
```diff
@@ -30,12 +30,33 @@
 #include <sstream>
 #include <sys/socket.h>
 
+#ifdef __APPLE__
+#include <libkern/OSByteOrder.h>
+#define htobe16(x) OSSwapHostToBigInt16(x)
```
Comment (Moandor-y) [inline src/common/types.cpp:35]: Avoid defining macros. Functions should be preferred here.

**Review by Moandor-y — COMMENTED** [review]
Most HIDL interfaces cannot be tested by CI. Should we have untested components in GitHub?

File: src/common/types.cpp (hunk 1 above)
Comment (zhanglongxia) [reply src/common/types.cpp:35]: Are there any special reasons to avoid defining macros?

**Comment by zhanglongxia** [comment]
> Most HIDL interfaces cannot be tested by CI. Should we have untested components in GitHub?
The HIDL interfaces are used in Android. When the code is synchronized to Android, they will be tested.

File: src/common/types.cpp (hunk 1 above)
Comment (Moandor-y) [reply src/common/types.cpp:35]: Macros don't have namespace, so they are likely to conflict with other names. Macros also have many well known pitfalls. And this is also included in [the style guide](https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros).

**Pull request closed without merging**